### ISSUE TITLE
XMLHttpRequest doesn't have headers

### ISSUE THREAD

**Issue by jslaybaugh** [jquery-form/form#110]
Have a situation where I'm posting a form and it returns either text/html or application/json based upon what happens.  So, I thought I'd look at the 3rd return parameter and use the getResponseHeader("content-type") method and check what was in that so that I could decide how to respond.  Well, if I use $.ajax, it works fine. The getResponseHeader and getAllResponseHeaders functions both return what you'd expect.  However with the jQuery.Form plugin, both of those functions return undefined.  Not sure what is happening under the covers on the XMLHttpRequest object, but I think something is getting lost in the shuffle. 

Thanks!

**Comment by malsup** [comment]
Are you uploading a file?  If so, response headers are not available.

**Comment by jslaybaugh** [comment]
Yes, I am.  That's one of the reasons I can't (well at least not very easily) just do $.ajax.  So its actually by design and no response headers can be available at all?

**Comment by malsup** [comment]
Well, it's not intentional, it's a byproduct of the implementation.  And the
implementation is meant to support more than just 2011 browsers.  When
uploading files the form is submitted via a native browser post, and the
response is targeted to an iframe.  The response is then extracted from the
iframe and passed along to callbacks etc, as if it was a "typical" ajax
request.  But there is no way to access response headers with that approach.
 I will be adding support for FormData posting soon, which alleviates this
problem, but that will only be available in very modern browsers.

**Comment by jslaybaugh** [comment]
Very interesting.  I knew it used the iframe but hadn't dug into how -- because it just worked :).  I'll find another workaround.  Thanks for the prompt explanation.
